src/optimizer/rule/simplification.rs

```diff
@@ -109,6 +119,99 @@ impl Rule for SimplifyFilter {
     }
 }
 
+pub struct LikeRewrite;
+
+impl Rule for LikeRewrite {
+    fn pattern(&self) -> &Pattern {
+        &LIKE_REWRITE_RULE
+    }
+
+    fn apply(&self, node_id: HepNodeId, graph: &mut HepGraph) -> Result<(), OptimizerError> {
+        if let Operator::Filter(mut filter_op) = graph.operator(node_id).clone() {
+            if let ScalarExpression::Binary {
+                op: BinaryOperator::Like,
+                ref mut left_expr,
+                ref mut right_expr,
+                ty,
+            } = filter_op.predicate.clone()
+            {
+                if let ScalarExpression::Constant(value) = right_expr.as_ref() {
```

Comment:
`if let ScalarExpression::Constant(DataValue::Utf8(Some(value))) = right_expr.as_mut()` 

Instead of pattern matching layer by layer again